docs/intro-to-coalton.md

```diff
@@ -4,15 +4,63 @@ Coalton is a statically typed language that is embedded in, and compiles to, Com
+Unlike Common Lisp, Coalton's standard library is organized as a large collection of packages. For example, string-related functions are in the `#:coalton-library/string` package. Refer to the [Coalton Reference](https://coalton-lang.github.io/reference) for a complete list of standard library packages.
+
+If you're playing around with Coalton in the REPL, we recommend changing your package to `#:coalton-user` like so:
 
 ```lisp
 (in-package #:coalton-user)
 ```
 
-This package does *not* `:use` the `COMMON-LISP` package, so you must prepend Common Lisp symbols with `cl:` if you need them.
+This is a convenience package that is defined purely for your interactive and experimental use. (No software should ever depend on it, just like `#:cl-user`.)
+
+When making a new project, you will want to define a new package and establish your standard library imports. *Do not `:use` the `#:cl` or `#:common-lisp` packages!* Instead, you'll want to, at minimum, `:use` both the `#:coalton` package (for core language features) and `#:coalton-prelude` (for extremely common standard library definitions):
+
+```lisp
+(defpackage #:my-package
+  (:use #:coalton #:coalton-prelude))
+```
+
+The `#:coalton-prelude` package does not contain any functionality that isn't also found elsewhere in the standard library, and hence, it is a convenience.
+
+In idiomatic usage, this minimal package definition would be far too limiting to be useful. For instance, if we wanted to use the standard library function `strip-prefix` (which strips a prefix off of a string), we would have to type `coalton-library/string:strip-prefix` in our program. Instead, we can create a local nickname for this package:
+
+```lisp
+(defpackage #:my-package
+  (:use #:coalton)
+  (:local-nicknames (#:str #:coalton-library/string)))
+```
+
+This way, we can now just type `str:strip-prefix`. Any time we make use of a new function, either from the Coalton standard library or from a third-party library, we might add nicknames that function's package to our `:local-nicknames` list. For example:
+
+```lisp
+(defpackage #:my-package
+  (:use #:coalton)
+  (:local-nicknames (#:str  #:coalton-library/string)
+                    (#:vec  #:coalton-library/vector)
+                    (#:math #:coalton-library/math)))
+```
+
+**We do not recommend `:use`ing any other package besides `#:coalton` and `#:coalton-prelude`.** Historically, in Common Lisp, this has caused backwards-compatibility issues when the packages themselves introduce new exported symbols. Moreover, unlike Common Lisp, Coalton follows a pattern of using the same symbol name for similarly defined functions of different packages. For example, both the string package and the vector package have a symbol named `length`, i.e., `str:length` and `vec:length` both exist.
+
+
+## Program Structure
+
+After creating your package `#:my-package`, you must switch to it with:
+
+```lisp
+(named-readtables:in-readtable coalton:coalton)
```

Comment:
It seems like the library tends to put `in-package` before `in-readtable`. It doesn't matter, but for consistency's sake we might as well use the same order in the example as in the source.

Reply:
I thought about this a little. I like consistency, but then I thought: If you're going to change the reader, why not do that as soon as possible, so you don't have "pre-reader" code and "post-reader" code?

Reply:
Eh, I could go either way. I'm not sure an `in-package` form counts as "code," and I think it's nice that the `defpackage` and `in-package` go right next to each other. But I don't think it really matters, just that there's some small value in having the docs be consistent with the library.

Reply:
`in-readtable` should happen after `in-package` so that the editor hints with SWANK can associate the readtable with the package for the file

Reply:
im fine to change it since the stdlib is that way, but I don't personally think it's "correct"